Add a test for search_form

cjmayo opened the pull request:
Just as well, this has been broken since 51a06d8a. Also commits to fix that and a problem message character.

mgedmin commented on tests/test_formsearch.py:
```diff
@@ -0,0 +1,49 @@
+<input name="some_field">
+</form>
+<h1>Login</h1>
+<form action="https://log.me.in">
```
I would feel slightly better if we used [IANA-reserved](https://www.iana.org/domains/reserved) domain names like https://notlogin.example.com and https://login.example.com for example URLs in tests.

cjmayo replied:
OK, good point. Actually these could just be relative locations - shall we hold off merging until updated?

mgedmin replied:
Either way is fine for me.

(BTW good idea on the relative locations bit!)

cjmayo commented:
Changed to relative locations, added another input field to the login form and another test for the case when no login fields are found:

```diff
--- a/tests/test_formsearch.py
+++ b/tests/test_formsearch.py
@@ -25,13 +25,14 @@ login_form = """
 <html>
 <body>
 <h1>Not a Login</h1>
-<form action="https://not.a.login">
+<form action="/not_a_login">
 <input name="some_field">
 </form>
 <h1>Login</h1>
-<form action="https://log.me.in">
+<form action="/log_me_in">
 <input name="user_field">
 <input name="password_field">
+<input name="extra_field">
 </form>
 </body>
 </html>
@@ -44,6 +45,11 @@ class TestFormSearch(unittest.TestCase):
         form = formsearch.search_form(login_form,
                                       "user_field", "password_field")
         self.assertIsNotNone(form)
-        self.assertEqual(form.url, "https://log.me.in")
+        self.assertEqual(form.url, "/log_me_in")
         self.assertIn("user_field", form.data)
         self.assertIn("password_field", form.data)
+
+    def test_search_form_none(self):
+        form = formsearch.search_form(login_form,
+                                      "nouser", "nopassword")
+        self.assertIsNone(form)
```
